core/unit_test/hip/TestHIP_InterOp_Graph.cpp

```diff
@@ -71,10 +71,12 @@ TEST(TEST_CATEGORY, graph_promises_on_native_objects) {
 TEST(TEST_CATEGORY, graph_instantiate_and_debug_dot_print) {
 #if !defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)
   GTEST_SKIP() << "This test will not work without native graph support";
-#elif KOKKOS_COMPILER_GNU < 910
-  GTEST_SKIP() << "'filesystem' is not fully supported prior to GCC 9.1.0.";
-#elif KOKKOS_COMPILER_CLANG < 1100
-  GTEST_SKIP() << "'filesystem' is not fully supported prior to LLVM 11.";
+#elif defined(KOKKOS_COMPILER_GNU) && KOKKOS_COMPILER_GNU < 910
+  GTEST_SKIP()
+      << "GNU implementation prior to 9.1 requires linking with '-lstdc++fs'.";
+#elif defined(KOKKOS_COMPILER_CLANG) && KOKKOS_COMPILER_CLANG < 1100
+  GTEST_SKIP() << "LLVM implementation prior to LLVM 11.0 requires linking "
+                  "with '-lc++fs'.";
```

Comment:
An issue I'm running across is that clang16 on the machine I'm running with requires this link as well, even though it is version compliant. It looks like it is because I am  having to use an older versions of the GNU standard library libstdc++ (gcc 8.5).

Reply:
I changed the guarding conditions to compare the `libstdc++` version instead of the compiler. Does it help ?

Reply:
No, this did not work. I think that we cant use `__GLIBCXX__`. Here's what I get
```
[tccleve@fat1 unit_test]$ gcc --version
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
I'm not sure, but I'm guessing that 8.5 could have been updated after releases that are compliant?

Reply:
Yeah looks like that's right, we can't use `__GLIBCXX__`. GCC continues to add minor updates to older major versions and `__GLIBCXX__` just is the release date.

Reply:
What about `_GLIBCXX_RELEASE`?

Reply:
Yeah, let's use `_GLIBCXX_RELEASE` as it's done at other locations in the code base.